// react-native-svga-player/index.js
// import { Platform } from "react-native";


// export default SVGAPlayer = Platform.OS === "ios"
//     ? require('./index.ios').default
//     : require('./index.android').default;

import React from 'react';
import { requireNativeComponent, NativeModules } from 'react-native';
const RNSvgaPlayer = requireNativeComponent('RNSvgaPlayer');
export default class SVGAPlayer extends React.Component {
    constructor(props) {
        super(props);
        this.state = {};
    }
    load(source) {
        this.setState({
            source,
        });
    }
    startAnimation() {
        this.setState({
            currentState: 'start',
        });
    }
    pauseAnimation() {
        this.setState({
            currentState: 'pause',
        });
    }
    stopAnimation() {
        this.setState({
            currentState: 'stop',
        });
    }
    stepToFrame(toFrame, andPlay) {
        this.setState({
            currentState: andPlay === true ? 'play' : 'pause',
            toFrame: -1,
        }, () => {
            this.setState({
                toFrame,
            });
        });
    }
    stepToPercentage(toPercentage, andPlay) {
        this.setState({
            currentState: andPlay === true ? 'play' : 'pause',
            toPercentage: -1,
        }, () => {
            this.setState({
                toPercentage,
            });
        });
    }
    componentWillUnmount() {
        this.stopAnimation();
    }
    render() {
        if (!this.props.source) {
            return null;
        }
        return <RNSvgaPlayer {...this.props} {...this.state}/>;
    }
}

export const prefetch = NativeModules.RNSvgaPlayer.prefetch;
